builtin/docker/platform.go

```diff
@@ -126,65 +130,126 @@ func (p *Platform) Status(
 	s := sg.Add("Gathering health report for Docker platform...")
 	defer s.Abort()
 
-	// currently the docker platform only deploys 1 container
-	containerInfo, err := cli.ContainerInspect(ctx, deployment.Container)
-	if err != nil {
-		return nil, err
-	}
+	log.Debug("querying docker for container health")
 
-	// Create our status report
-	var result sdk.StatusReport
-	result.External = true
+	// Creating our baseline container resource
+	containerResource := &sdk.StatusReport_Resource{
+		Type:                "container",
+		Platform:            platformName,
+		CategoryDisplayHint: sdk.ResourceCategoryDisplayHint_INSTANCE,
+	}
 
 	// NOTE(briancain): The docker platform currently only deploys a single
 	// container, so for now the status report makes the same assumption.
+	containerInfo, err := cli.ContainerInspect(ctx, deployment.Container)
+	if err != nil {
+		if client.IsErrNotFound(err) {
+			// We expected this container to be present, but it's not.
+			// It has likely been killed and removed from the platform since it was deployed.
+			containerResource.Name = deployment.Container
+			containerResource.Health = sdk.StatusReport_MISSING
+		} else {
+			return nil, fmt.Errorf("error quering docker for container status: %s", err)
```

Comment:
We should use the `codes/status` errors here too, probably an failed precondition error?

Reply:
Done